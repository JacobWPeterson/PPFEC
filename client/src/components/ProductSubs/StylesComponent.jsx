import React from 'react';
import styled from 'styled-components';

const StyleWrapper = styled.div`
  margin-bottom: 10px;
`;

const TextWrapper = styled.div`
  display: flex;
  flex-direction: row;
`;

const Style = styled.p`
  font-weight: 700;
  color: #424242;
`;

const SelectedStyle = styled.p`
  margin-left: 1vw;
  font-weight: 300;
  color: #424242;
`;

const Thumbs = styled.div`
flex-basis: 90%
display: flex;
  flex-wrap: wrap;
  margin: 0;
`;

const Image = styled.img`
  flex: 1 0 22%;
  margin: 1vh 1.2vw 1vh 0;
  border: 2px solid #aeaeae;
  border-radius: 100%;
  z-index: 0;
`;

const SelectedImage = styled.img`
  flex: 1 0 22%;
  margin: 1vh 1.2vw 1vh 0;
  border: 3px solid #80ccc4;
  border-radius: 100%;
  z-index: 0;
`;

class Styles extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentStyle: null,
      style_id: null,
    };
    this.clickHandler = this.clickHandler.bind(this);
    this.passUpStyle = this.passUpStyle.bind(this);
  }

  componentDidMount() {
    this.setState({
      currentStyle: this.props.styles.results[0].name,
      style_id: this.props.styles.results[0].style_id,
    });
  }

  clickHandler(event) {
    event.preventDefault();
    this.setState({
      currentStyle: event.target.alt,
      style_id: event.target.id,
    }, () => this.passUpStyle(this.state.style_id));
  }

  passUpStyle(style) {
    this.props.getStyleID(style);
  }

  render() {
    if (this.state.currentStyle) {
      return (
        <StyleWrapper>
          <TextWrapper>
            <Style>STYLE &gt;</Style>
            <SelectedStyle>{this.state.currentStyle.toUpperCase()}</SelectedStyle>
          </TextWrapper>
          <Thumbs>
            { this.props.styles.results.map((image) => {
              if (image.style_id === this.state.style_id) {
                return <SelectedImage onClick={this.clickHandler} key={image.style_id} id={image.style_id} src={image.photos[0].thumbnail_url} alt={image.name} width="65" height="65" />;
              }
              if (image.style_id !== this.state.style_id) {
                return <Image onClick={this.clickHandler} key={image.style_id} id={image.style_id} src={image.photos[0].thumbnail_url} alt={image.name} width="65" height="65" />;
              }
              return <div>Hello</div>;
            })}
          </Thumbs>
        </StyleWrapper>
      );
    }
    return <div>Empty</div>;
  }
}

export default Styles;
