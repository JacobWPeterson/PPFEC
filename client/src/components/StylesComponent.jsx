import React from 'react';

const Styles = () => (
  <div>
    <div>Style</div>
    <div>Selected Style</div>
    <div>Style thumbnails</div>
  </div>
);

export default Styles;
