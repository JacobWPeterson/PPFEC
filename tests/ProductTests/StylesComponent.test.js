/* eslint-disable no-undef */
/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { mount } from 'enzyme';

import StylesComponent from '../../client/src/components/ProductSubs/StylesComponent.jsx';

const props = {
  styles: {
    product_id: '14036',
    results: [
      {
        style_id: 70550, name: 'Black', original_price: '40.00', sale_price: null, photos: [{ thumbnail_url: 'https://images.unsplash.com/photo-1552902865-b72c031ac5ea?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80' }],
      },
      {
        style_id: 70551, name: 'Grey', original_price: '40.00', sale_price: null, photos: [{ thumbnail_url: 'https://images.unsplash.com/photo-1562542082-519ebcdb43e6?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80' }],
      },
      {
        style_id: 70552,
        name: 'Goldenrod',
        original_price: '40.00',
        sale_price: '35.00',
        photos: [{ thumbnail_url: 'https://images.unsplash.com/photo-1530073391204-7b34a1497281?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80' }],
      },
      {
        style_id: 70553, name: 'Maroon', original_price: '40.00', sale_price: '35.00', photos: [{ thumbnail_url: 'https://images.unsplash.com/photo-1519857609704-61e751edba25?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80' }],
      },
      {
        style_id: 70554, name: 'Chartreuse', original_price: '40.00', sale_price: '25.00', photos: [{ thumbnail_url: 'https://images.unsplash.com/photo-1530073391204-7b34a1497281?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80' }],
      },
      {
        style_id: 70555, name: 'White', original_price: '40.00', sale_price: null, photos: [{ thumbnail_url: 'https://images.unsplash.com/photo-1510390099355-23e690d8129d?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80' }],
      }],
  },
};

describe('StylesComponent', () => {
  const StylesComponentComponent = mount(<StylesComponent {...props} debug />);

  it('Should render correctly', () => {
    expect(StylesComponentComponent).toMatchSnapshot();
  });
});
