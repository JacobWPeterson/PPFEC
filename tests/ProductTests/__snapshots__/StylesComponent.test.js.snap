// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StylesComponent Should render correctly 1`] = `
<Styles
  debug={true}
  styles={
    Object {
      "product_id": "14036",
      "results": Array [
        Object {
          "name": "Black",
          "original_price": "40.00",
          "photos": Array [
            Object {
              "thumbnail_url": "https://images.unsplash.com/photo-1552902865-b72c031ac5ea?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
            },
          ],
          "sale_price": null,
          "style_id": 70550,
        },
        Object {
          "name": "Grey",
          "original_price": "40.00",
          "photos": Array [
            Object {
              "thumbnail_url": "https://images.unsplash.com/photo-1562542082-519ebcdb43e6?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
            },
          ],
          "sale_price": null,
          "style_id": 70551,
        },
        Object {
          "name": "Goldenrod",
          "original_price": "40.00",
          "photos": Array [
            Object {
              "thumbnail_url": "https://images.unsplash.com/photo-1530073391204-7b34a1497281?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
            },
          ],
          "sale_price": "35.00",
          "style_id": 70552,
        },
        Object {
          "name": "Maroon",
          "original_price": "40.00",
          "photos": Array [
            Object {
              "thumbnail_url": "https://images.unsplash.com/photo-1519857609704-61e751edba25?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
            },
          ],
          "sale_price": "35.00",
          "style_id": 70553,
        },
        Object {
          "name": "Chartreuse",
          "original_price": "40.00",
          "photos": Array [
            Object {
              "thumbnail_url": "https://images.unsplash.com/photo-1530073391204-7b34a1497281?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
            },
          ],
          "sale_price": "25.00",
          "style_id": 70554,
        },
        Object {
          "name": "White",
          "original_price": "40.00",
          "photos": Array [
            Object {
              "thumbnail_url": "https://images.unsplash.com/photo-1510390099355-23e690d8129d?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
            },
          ],
          "sale_price": null,
          "style_id": 70555,
        },
      ],
    }
  }
>
  <StylesComponent__StyleWrapper>
    <div
      className="StylesComponent__StyleWrapper-ajk3ts-0 bQXBFa"
    >
      <Price
        price="40.00"
        sale={null}
      >
        <Price__NoSale>
          <p
            className="Price__NoSale-eho440-0 ituvCN"
          >
            $40
          </p>
        </Price__NoSale>
      </Price>
      <StylesComponent__TextWrapper>
        <div
          className="StylesComponent__TextWrapper-ajk3ts-1 kDJaHh"
        >
          <StylesComponent__Style>
            <p
              className="StylesComponent__Style-ajk3ts-2 glNYdS"
            >
              STYLE &gt;
            </p>
          </StylesComponent__Style>
          <StylesComponent__SelectedStyle>
            <p
              className="StylesComponent__SelectedStyle-ajk3ts-3 RbxlX"
            >
              BLACK
            </p>
          </StylesComponent__SelectedStyle>
        </div>
      </StylesComponent__TextWrapper>
      <StylesComponent__Thumbs>
        <div
          className="StylesComponent__Thumbs-ajk3ts-4 gBVoCt"
        >
          <StylesComponent__SelectedImage
            alt="Black"
            id={70550}
            key="70550"
            onClick={[Function]}
            src="https://images.unsplash.com/photo-1552902865-b72c031ac5ea?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
          >
            <img
              alt="Black"
              className="StylesComponent__SelectedImage-ajk3ts-6 eDgpFT"
              id={70550}
              onClick={[Function]}
              src="https://images.unsplash.com/photo-1552902865-b72c031ac5ea?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
            />
          </StylesComponent__SelectedImage>
          <StylesComponent__Image
            alt="Grey"
            id={70551}
            key="70551"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            src="https://images.unsplash.com/photo-1562542082-519ebcdb43e6?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
          >
            <img
              alt="Grey"
              className="StylesComponent__Image-ajk3ts-5 cdIpah"
              id={70551}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              src="https://images.unsplash.com/photo-1562542082-519ebcdb43e6?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
            />
          </StylesComponent__Image>
          <StylesComponent__Image
            alt="Goldenrod"
            id={70552}
            key="70552"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            src="https://images.unsplash.com/photo-1530073391204-7b34a1497281?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
          >
            <img
              alt="Goldenrod"
              className="StylesComponent__Image-ajk3ts-5 cdIpah"
              id={70552}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              src="https://images.unsplash.com/photo-1530073391204-7b34a1497281?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
            />
          </StylesComponent__Image>
          <StylesComponent__Image
            alt="Maroon"
            id={70553}
            key="70553"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            src="https://images.unsplash.com/photo-1519857609704-61e751edba25?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
          >
            <img
              alt="Maroon"
              className="StylesComponent__Image-ajk3ts-5 cdIpah"
              id={70553}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              src="https://images.unsplash.com/photo-1519857609704-61e751edba25?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
            />
          </StylesComponent__Image>
          <StylesComponent__Image
            alt="Chartreuse"
            id={70554}
            key="70554"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            src="https://images.unsplash.com/photo-1530073391204-7b34a1497281?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
          >
            <img
              alt="Chartreuse"
              className="StylesComponent__Image-ajk3ts-5 cdIpah"
              id={70554}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              src="https://images.unsplash.com/photo-1530073391204-7b34a1497281?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
            />
          </StylesComponent__Image>
          <StylesComponent__Image
            alt="White"
            id={70555}
            key="70555"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            src="https://images.unsplash.com/photo-1510390099355-23e690d8129d?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
          >
            <img
              alt="White"
              className="StylesComponent__Image-ajk3ts-5 cdIpah"
              id={70555}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              src="https://images.unsplash.com/photo-1510390099355-23e690d8129d?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
            />
          </StylesComponent__Image>
        </div>
      </StylesComponent__Thumbs>
    </div>
  </StylesComponent__StyleWrapper>
</Styles>
`;
